lib/maths/CBoostedTreeImpl.cc

```diff
@@ -203,6 +203,8 @@ void CBoostedTreeImpl::train(core::CDataFrame& frame,
 
     this->startProgressMonitoringFineTuneHyperparameters();
 
+    this->initializeHyperparameterSamples();
```

Comment:
nit: logically this seems to belong right before `while (m_CurrentRound < m_NumberRounds) {` together with call to `this->initializePerFoldTestLosses();`.